src/DependencyInjection/Compiler/RoutePass.php

```diff
@@ -62,22 +73,27 @@ public function process(ContainerBuilder $container)
                     }
                 }
                 $router->setArguments($routerArguments);
+
+                // Update route configuration parameter
+                $configId = sprintf('prooph_service_bus.%s.configuration', $name);
+
+                $config = array_replace($container->getParameter($configId), ['router' => ['routes' => $routerArguments[0]]]) ;
+
+                $config = $container->setParameter($configId, $config);
             }
         }
     }
 
-    private function recognizeMessageNames(ContainerBuilder $container, $id, array $args)
+    private function recognizeMessageNames(Definition $routeDefinition, array $args): array
     {
-        if (isset($args['message'])) {
-            return [$args['message']];
-        }
-        $handlerReflection = new ReflectionClass($container->getDefinition($id)->getClass());
+        $handlerReflection = new ReflectionClass($routeDefinition->getClass());
```

Comment:
Symfony 3.3 added a method `ContainerBuilder::getReflectionClass()` for reflection that also takes care of resource tracking. https://github.com/symfony/symfony/pull/21419